src/Shared/ProjectWriter.cs

```diff
@@ -118,8 +118,8 @@ internal void Initialize(XmlDocument project, XmlDeclaration projectRootElementD
             }
 
             // don't write an XML declaration unless the project already has one or has non-default encoding
-            _writeXmlDeclaration = (projectRootElementDeclaration != null) ||
-                                   (_documentEncoding != null && !Equals(_documentEncoding, Encoding.UTF8));
+            _writeXmlDeclaration = projectRootElementDeclaration != null ||
```

Comment:
Unrelated to PR: should the first expression be `projectRootElementDeclaration == null`? a.k.a write the xml declaration if there is no existing one.